src/components/Text2Image/Text2ImageDisplay.vue

```diff
@@ -0,0 +1,436 @@
+			// (as we don't know when the scheduled job will start exactly)
+			if (timeDifference < 5 * 60000) {
+				this.timeUntilCompletion = null
+				return
+			}
+
+			this.timeUntilCompletion = humanizeDuration(timeDifference,
+				{ units: ['h', 'm'], language: OC.getLanguage(), fallbacks: ['en'], round: true })
+
+			// Schedule next update:
+			if (!this.closed) {
+				setTimeout(() => {
+					this.updateTimeUntilCompletion(completionTimeStamp)
+				}, 30000)
+			}
+		},
+		onError(error) {
+			// If error response status is 429 let the user know that they are being rate limited
+			if (error.response?.status === 429) {
+				this.errorMsg = t('assistant', 'Rate limit reached. Please try again later.')
+				this.failed = true
+				this.imgLoadedList = []
+			} else if (error.response?.data !== undefined) {
+				this.errorMsg = error.response.data.error
+				this.failed = true
+				this.imgLoadedList = []
+			} else {
+				console.error('Could not handle response error: ' + error)
+				this.errorMsg = t('assistant', 'Unknown server query error')
+				this.failed = true
+				this.imgLoadedList = []
+			}
+			this.$emit('failed')
+
+		},
+		onGenerationReady() {
+			this.$emit('ready')
+		},
+		onCheckboxChange() {
+			const url = generateUrl('/apps/assistant/i/visibility/' + this.src.split('/').pop())
+
+			axios.post(url, {
+				fileVisStatusArray: this.fileVisStatusArray,
+			})
+				.then((response) => {
+					if (response.status === 200) {
+						// console.log('Successfully updated visible images')
+					} else {
+						console.error('Unexpected response status: ' + response.status)
+					}
+				})
+				.catch((error) => {
+					console.error('Could not update visible images: ' + error)
+				})
+		},
+		toggleCheckbox(index) {
+			this.fileVisStatusArray[index].visible = !this.fileVisStatusArray[index].visible
+			this.onCheckboxChange()
+		},
+		toggleEditMode() {
+			this.editModeEnabled = !this.editModeEnabled
+		},
+	},
+}
+</script>
+
+<style scoped lang="scss">
+.display-container {
+	display: flex;
+	flex-direction: column;
+	width: 100%;
+	align-items: center;
+	justify-content: center;
+	.edit-icon {
+		position: static;
+		opacity: 0.2;
+		transition: opacity 0.2s ease-in-out;
+		cursor: pointer;
+	}
+
+	.edit-icon.active {
+		opacity: 1;
+		cursor: pointer;
+	}
+
+	.image-list {
+		display: flex;
+		flex-direction: column;
+		flex-wrap: wrap;
+		justify-content: center;
+	}
+
+	.image-container {
+		display: flex;
+		flex-direction: column;
+		position: relative;
+		justify-content: center;
+		max-width: 90%;
+	}
+
+	.checkbox-container {
+		position: absolute;
+		top: 5%;
+		left: 95%;
+		z-index: 1;
+		opacity: 0.2;
+		transition: opacity 0.2s ease-in-out;
+	}
+
+	.checkbox-container.hovering {
+		opacity: 1;
+	}
+
+	.image-wrapper {
+		display: flex;
+		flex-direction: column;
+		position: relative;
+		max-width: 100%;
+		height: 100%;
+		margin-top: 12px;
+		filter: grayscale(100%);
+		transition: filter 0.2s ease-in-out;
```

Comment:
This does not work on my side and the images stay gray everywhere they get displayed (preview in the picker, link preview widget).

Reply:
Hmm.. That's unfortunate. On Firefox, or?

Reply:
Note to self: filter should be none for the default state.